storage/database/interface.go

```diff
@@ -93,6 +93,38 @@ func WriteBatches(batches ...Batch) (int, error) {
 	return bytes, nil
 }
 
+func WriteBatchesParallel(batches ...Batch) (int, error) {
+	type result struct {
+		bytes int
+		err   error
+	}
+
+	resultCh := make(chan result, len(batches))
+	for _, batch := range batches {
+		go func(batch Batch) {
+			bytes := batch.ValueSize()
+			err := batch.Write()
+			if err != nil {
+				batch.Reset()
+			}
+			resultCh <- result{bytes, err}
+		}(batch)
+	}
+
+	var bytes int
+	var errResult error
+
+	for range batches {
+		rst := <-resultCh
+		if rst.err != nil {
+			errResult = rst.err
```

Comment:
Don't we need to leave a log for each error? Since only the last error will be returned and logged.

Reply:
I think it's ok to know one error even if there are several errors.
But if we know all error. How about this code?
But I think I don't need this code. What do you think? :)
```suggestion
			errResult = fmt.Errorf("%v - %v", errResult, rst.err)
```

Reply:
If so, we can immediately return the error once we receive it. As batch write will be done anyway as they are executed by go-routine. Also the channel will be garbage-collected. We don't need to wait to check the whole result.
```suggestion
			return rst.err
```

Reply:
Good point. I reflected. :)